bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -250,6 +253,10 @@ public BigtableSession(BigtableOptions opts) throws IOException {
     dataClient =
         new BigtableDataGrpcClient(dataChannel, sharedPools.getRetryExecutor(), options);
     dataClient.setCallOptionsFactory(callOptionsFactory);
+    BigtableDataSettings dataSettings =
```

Comment:
Please delay creating the `dataSettings` and `dataGCJClient` until the client is actually needed.